### PR TITLE
Show in Web UI if file is vendored and generated

### PR DESCRIPTION
**Pull request by JakobDev**
This simple shows in the Web UI is a file is vendored and/or generated.

grafik
grafik

### CODE REVIEW

**Comment by lunny** [comment]
There is a similar logic on the compare page. Maybe the template can be shared.

File: routers/web/repo/view.go [hunk 1]
```diff
@@ -596,6 +596,25 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		}
 	}
 
+	if ctx.Repo.GitRepo != nil {
+		checker, deferable := ctx.Repo.GitRepo.CheckAttributeReader(ctx.Repo.CommitID)
```
Comment (wxiaoguang) [inline routers/web/repo/view.go:651]: You need to check `nil`, otherwise it could panic.

File: templates/repo/file_info.tmpl [hunk 2]
```diff
@@ -30,6 +30,16 @@
 			{{ctx.Locale.Tr "repo.executable_file"}}
 		</div>
 	{{end}}
+	{{if .IsVendored}}
+		<div class="file-info-entry">
+			{{ctx.Locale.Tr "repo.vendored"}}
```
Comment (wxiaoguang) [inline templates/repo/file_info.tmpl:35]: `repo.vendored` doesn't seem a good key

Actually, you could reuse the `repo.diff.vendored` and reuse that UI.

Comment (JakobDev) [reply templates/repo/file_info.tmpl:35]: These strings are lowercase. Should I make it uppercase to match the rest of the Header?

**Comment by JakobDev** [comment]
> There is a similar logic on the compare page. Maybe the template can be shared.

The code is [here](https://github.com/go-gitea/gitea/blob/cb10f2767c920fdbca6fc43e71dcc4c4b8c6e0d0/services/gitdiff/gitdiff.go#L1187) and is part of a larger function. I don't think it makes much sense to introduce a new service function for this few simple lines. For the template, it should look different as they are different usecases.

File: templates/repo/file_info.tmpl (hunk 2 above)
Comment (wxiaoguang) [reply templates/repo/file_info.tmpl:35]: > reuse that UI.

What do you think about this? Reuse that UI and reuse those texts.

Comment (JakobDev) [reply templates/repo/file_info.tmpl:35]: The UI looks a little bit different than the UI on the view file page.

File: routers/web/repo/view.go [hunk 3]
```diff
@@ -596,6 +596,27 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
...
+					if vendored == "set" || vendored == "true" {
+						ctx.Data["IsVendored"] = true
+					}
+				}
+
+				if generated, has := attrs["linguist-generated"]; has {
+					if generated == "set" || generated == "true" {
+						ctx.Data["IsGenerated"] = true
+					}
+				}
```
Comment (delvh) [inline routers/web/repo/view.go]:
```suggestion
				vendored, has := attrs["linguist-vendored"]
				ctx.Data["IsVendored"] = has && (vendored == "set" || vendored == "true")
				
				generated, has := attrs["linguist-generated"]
				ctx.Data["IsGenerated"] = has && (generated == "set" || generated == "true")
```

**Review by delvh — APPROVED** [review]

**Review by denyskon — APPROVED** [review]
(assuming the changes proposed by delvh)

**Pull request merged**